lighthouse-core/gather/gather-runner.js

```diff
@@ -172,6 +173,22 @@ class GatherRunner {
     }
   }
 
+  /**
+   * Returns an error if the original network request failed or wasn't found.
+   * @param {LH.Crdp.Security.SecurityStateChangedEvent} securityState
+   * @return {LHError|undefined}
+   */
+  static checkForSecurityIssue({securityState, explanations}) {
+    if (securityState === 'insecure') {
+      const errorDef = {...LHError.errors.INSECURE_DOCUMENT_REQUEST};
+      const insecureDescriptions = explanations
+        .filter(exp => exp.securityState === 'insecure')
+        .map(exp => exp.description);
+      errorDef.message += ` ${insecureDescriptions.join(' ')}`;
+      return new LHError(errorDef, {fatal: true});
```

Comment:
rather than returning the error and throwing on the return value, I'd say just throw in here

Reply:
`{fatal: true}` shouldn't be necessary anymore

Reply:
OK, I can do that. I was thinking it's nicer to make tests that check return values rather than throwing conditions, but just my preference, I could have it either way.

I'll remove fatal. yay, that means the resolve or throw thing really can still be removed.